### DISCUSSION TITLE
pest practice for newline at string

### DISCUSSION THREAD

**Discussion by LuckyTurtleDev** [DioxusLabs/dioxus#2923] [Q&A] [answered]
hi, I would like to display a string at dioxus. The string include newline. What is the pest practice for displaying the newline.
Exist a better option as spitting the string at `\n` into multiple elements and adding `<br>` between it?

**Comment by ealmloff** [comment, marked as answer]
You can use the [pre element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/pre) if you need to display preformatted text

**Reply by LuckyTurtleDev** [reply]
thanks,
learned something new again.

tipp:
To limit the  width of the `<pre>` ` white-space: pre-wrap;` should be set.
```css
pre {
    white-space: pre-wrap;
    max-width: min(65em, 98%) ;
}
```

**Reply by eranfu** [reply]
Hi, I tried this as well, but it didn't work for me. My situation might be slightly different because the content is a dynamically generated string converted from an error.

Please help me. Thank you~~

The code:
```
rsx! {
            match result {
                Ok(_) => {
                    rsx! {
                    }
                }
                Err(e) => {
                    rsx! {
                        pre {
                            "Error: {e}"
                        }
                    }
                }            
        }
    }
```
